impl/src/main/java/org/ehcache/internal/classes/ClassInstanceProvider.java

```diff
@@ -76,7 +80,7 @@ private T newInstance(String alias, ClassInstanceConfiguration<T> config) {
       }
     }
     try {
-      return invokeConstructor(config.getClazz(), config.getArguments());
+      return invokeConstructor(config.getClazz(), args.length > 0 ? args : config.getArguments());
```

Comment:
The contract here is enforced by the two protected method calling in, i.e. either
 - the config passed in was null, so we _ignore_ possible preconfigured config args
 - the vararg was empty, so we use the config (preconfigured or not)

Not sure how happy I am with this contract